src/embeddings/tools/embedding_vocab.cpp

```diff
@@ -16,6 +16,28 @@
 
 using namespace meta;
 
+namespace
+{
+    const std::string SGNS  = "sgns";
+    const std::string GLOVE = "glove";
+}
+
+hashing::probe_map<std::string, uint64_t> reduce_vocab(hashing::probe_map<std::string, uint64_t> vocab,
+                                                       unsigned int min_reduce)
+{
```

Comment:
I meant to comment on this one before too. What you said sounds familiar, and I think that is indeed why we did this.